frontends/web/src/app.tsx

```diff
@@ -66,15 +67,16 @@ export const App = () => {
     const currentURL = window.location.pathname;
     const isIndex = currentURL === '/' || currentURL === '/index.html' || currentURL === '/android_asset/web/index.html';
     const inAccounts = currentURL.startsWith('/account/');
+    const noAccounts = (!accounts || accounts.length === 0) && lightningConfig.accounts.length === 0;
 
     // QT and Android start their apps in '/index.html' and '/android_asset/web/index.html' respectively
     // This re-routes them to '/' so we have a simpler uri structure
-    if (isIndex && currentURL !== '/' && (!accounts || accounts.length === 0)) {
+    if (isIndex && currentURL !== '/' && noAccounts) {
       navigate('/');
       return;
     }
     // if no accounts are registered on specified views route to /
-    if (accounts.length === 0 && (
+    if (noAccounts && (
```

Comment:
'add-account' and 'manage-accounts' pages should not depend on lightning accounts. I think that we need two different `noAccounts` conditions.

Reply:
please consider doing some of this refactoring if possible  on master.